packages/twenty-server/src/engine/api/rest/api-rest-query-builder/utils/fields.utils.ts

```diff
@@ -1,37 +1,56 @@
 import { BadRequestException } from '@nestjs/common';
 
+import { compositeTypeDefintions } from 'src/engine/metadata-modules/field-metadata/composite-types';
 import { FieldMetadataType } from 'src/engine/metadata-modules/field-metadata/field-metadata.entity';
+import { isCompositeFieldMetadataType } from 'src/engine/metadata-modules/field-metadata/utils/is-composite-field-metadata-type.util';
+import { ObjectMetadataEntity } from 'src/engine/metadata-modules/object-metadata/object-metadata.entity';
 import { computeObjectTargetTable } from 'src/engine/utils/compute-object-target-table.util';
 
 export const getFieldType = (
-  objectMetadataItem,
-  fieldName,
+  objectMetadata: ObjectMetadataEntity,
+  fieldName: string,
 ): FieldMetadataType | undefined => {
-  for (const itemField of objectMetadataItem.fields) {
-    if (fieldName === itemField.name) {
-      return itemField.type;
+  for (const fieldMetdata of objectMetadata.fields) {
+    if (fieldName === fieldMetdata.name) {
+      return fieldMetdata.type;
     }
   }
 };
 
-export const checkFields = (objectMetadataItem, fieldNames): void => {
+export const checkFields = (
+  objectMetadata: ObjectMetadataEntity,
+  fieldNames: string[],
+): void => {
+  const fieldMetadataNames = objectMetadata.fields
+    .map((field) => {
+      if (isCompositeFieldMetadataType(field.type)) {
+        const compositeType = compositeTypeDefintions.get(field.type);
+
+        if (!compositeType) {
+          throw new BadRequestException(
+            `Composite type '${field.type}' not found`,
+          );
+        }
+
+        // TODO: Don't really know why we need to put fieldName and compositeType name here
+        return [
+          field.name,
+          compositeType.properties.map(
+            (compositeProperty) => compositeProperty.name,
+          ),
+        ].flat();
+      }
+
+      return field.name;
+    })
+    .flat();
```

Comment:
@martmull If you can take a look on this please, I don't really understand why before you put `fieldMetadata.name` and `targetColumnMap` in the array, is there any specific reason ?

Reply:
Yes,
This check with fieldMetadata.name and targetColumnMap is performed when you filter on nested fields like FULL_NAME.

Here is an example of such a filter: `BASE_API_URL/rest/people?filter=name.firstName[startsWith]:Brian`

- `BASE_API_URL/rest/people?filter=naaaaaaaame.firstName[startsWith]:Brian` will return 
> "error": "BadRequestException: field 'naaaaaaaame' does not exist in 'person' object"

- `BASE_API_URL/rest/people?filter=name.firstNaaaaaaaame[startsWith]:Brian` will return 
> "error": "BadRequestException: field 'firstNaaaaaaaame' does not exist in 'person' object"

To do that you need to check that `name` and `firstName` are includes in `['name', 'firstName', 'lastName' ]` with `name` = `fieldMetadata.name` and `['firstName', 'lastName']` = `Object.keys(itemField.targetColumnMap)`

So you need to make sure that filter works for composite fields, and check that the error is explicite when a typo occurs on each subField (`name` or `firstName` in the above example)

Is that more clear?

Reply:
Ok I understand, my change should handle the following cases :) 
Thanks for the clarification !